src/move/visualization/vae_visualization.py

```diff
@@ -0,0 +1,267 @@
+__all__ = ["plot_vae"]
+
+from pathlib import Path
+from typing import Optional
+
+import matplotlib
+import matplotlib.cm as cm
+import matplotlib.figure
+import matplotlib.pyplot as plt
+import networkx as nx
+import numpy as np
+import torch
+
+
+def plot_vae(
```

Comment:
As a general hint: This function does too much in one place. I would also leave saving always to the main process (or have another flag for that)